### PR TITLE
add ed25519 to curve25519 conversion helpers

### LINKED ISSUE #204: Converting a verification key into public key
Hi,

I'm having the following working python code snippet in an an application
things works fine, however I wanted to convert  into Ruby using rbNaCL and am having an issue
```
sk = nacl.signing.SigningKey.generate().encode(nacl.encoding.Base64Encoder)
pk = sk.verify_key.to_curve25519_public_key().encode(encoder=nacl.encoding.Base64Encoder)

# send pk to user BOB where they encrypt data with pk and BoB private key
# then Alice recieves encrypted message with nonce and a public key that is base64 encoded
# `'LDuRaopE6MSgoJpswcSlXZsqqxaqv7LeDTgc4QbWQoQ='`
# then Alice decrypts using the following working code

userPubk = nacl.signing.VerifyKey('LDuRaopE6MSgoJpswcSlXZsqqxaqv7LeDTgc4QbWQoQ=', encoder=nacl.encoding.Base64Encoder)
box = Box(
        sk.to_curve25519_private_key(),
        userPubk.to_curve25519_public_key()
    )
decrypted = box.decrypt(ciphertext, nonce)
```
I want to achieve same thing with Ruby (I successfully did it in golang)
However I can not seem to be able to convert a key to `curve25519_public_key`
Here's a code snippet from the decryption phase

```
privateKey = RbNaCl::PrivateKey.generate
...........
............

# decryption phase
userPublicKey = Base64.strict_decode64('LDuRaopE6MSgoJpswcSlXZsqqxaqv7LeDTgc4QbWQoQ=')
pk = RbNaCl::PublicKey.new(userPublicKey)
userPublicKeyObj = RbNaCl::VerifyKey.new(pk)

decrypted = JSON.load(RbNaCl::Box.new(pk, privateKey).decrypt(nonce, cipherText))
```

my Issue is I can not seem to convert the public key into `curve25519` so the decrypt method fails
is there any way to achieve this? i.e
converting a ` #<RbNaCl::Signatures::Ed25519::VerifyKey:2c3b916a>` Object into
`RbNaCl::Boxes::Curve25519XSalsa20Poly1305::PublicKey:2c3b916a>`

### PR DESCRIPTION
**Pull request by abom**
Don't know if it's the right place to add these helpers or If I did any unintended mistakes or more tests need to be added!

If this need to be moved to `Util` module or somewhere else, just let me know.

https://libsodium.gitbook.io/doc/advanced/ed25519-curve25519 should be read for anyone who uses this (we already used this in one of our projects).

Resolves #204.

### CODE REVIEW

**Review by tarcieri — CHANGES_REQUESTED** [review]
The methods should at least indicate that they're converting from the Edwards ("Ed25519") to Montgomery ("Curve25519", "X25519") forms.

Unfortunately this gem uses the now outdated "Curve25519" naming which is a bit ambiguous (as Curve25519 is also often used to describe the curve regardless of what form it's expressed in).

All that said, maybe `to_curve25519_private_key` and `to_curve25519_public_key`?

**Comment by abom** [comment]
Yes, that would be less ambiguous for sure, updated method names and added `(x25519)` in the doc.

File: spec/rbnacl/signatures/ed25519/signing_key_spec.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@
     expect(subject.keypair_bytes).to eq signing_keypair
   end
 
+  it "can be converted to curve25519 private key" do
+    expect(subject.to_private_key).to be_a_kind_of(RbNaCl::PrivateKey)
+  end
```
Comment (tarcieri) [inline spec/rbnacl/signatures/ed25519/signing_key_spec.rb:36]: It'd be good to find some test vectors and ensure the output is correct for both methods

Comment (abom) [reply spec/rbnacl/signatures/ed25519/signing_key_spec.rb]: added test vectors (generated by `pynacl`).

**Review by tarcieri — APPROVED** [review]

**Comment by tarcieri** [comment]
Looks good, thanks!

**Comment by nlsrchtr** [comment]
Hi @tarcieri and @abom,

thanks for your work here. I was trying this on main branch with the following code:

```
signing_private_key = RbNaCl::Signatures::Ed25519::SigningKey.new(SecureRandom.random_bytes(32))
signing_public_key = signing_private_key.verify_key
private_key = signing_private_key.to_curve25519_private_key
public_key = private_key.public_key
```

and was running into `undefined method "to_curve25519_private_key" for #<RbNaCl::Signatures::Ed25519::SigningKey:eddda15d> (NoMethodError)`.

Is there anything I'm missing here? Thanks for your help!

**Comment by abom** [comment]
Hi @nlsrchtr 

I tested your example and it's working, please make sure you're using the correct version of `rbnacl`. I think you need to make sure it does load it from another path. Also, a complete backtrace would help.

**Pull request merged**